presto-main-base/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -887,6 +888,23 @@ public PlanWithProperties visitSpatialJoin(SpatialJoinNode node, StreamPreferred
             return rebaseAndDeriveProperties(node, ImmutableList.of(probe, build));
         }
 
+        @Override
+        public PlanWithProperties visitMergeJoin(MergeJoinNode node, StreamPreferredProperties parentPreferences)
+        {
+            PlanWithProperties probe = planAndEnforce(
+                    node.getLeft(),
+                    defaultParallelism(session),
```

Comment:
Can you double check if defaultParallelism is the ones to be used here, do we need to use singleStream here?

Reply:
I think we can hardcode `singleStream` to unblock the test. will take a deeper look what's the right way to handle parallelism here in follow-ups.